src/Symfony/Component/Yaml/Inline.php

```diff
@@ -287,6 +287,10 @@ public static function parseScalar($scalar, $delimiters = null, $stringDelimiter
                 throw new ParseException(sprintf('The reserved indicator "%s" cannot start a plain scalar; you need to quote the scalar.', $output[0]));
             }
 
+            if ($output && '%' === $output[0]) {
+                @trigger_error(sprintf('Not quoting a scalar starting with the "%s" indicator character is deprecated since Symfony 3.1 and will throw a ParseException in 4.0.', $output[0]), E_USER_DEPRECATED);
```

Comment:
You are right. I first thought to deprecate more indicator characters. But it turned out that all other were already handled properly.